apache/metron
# METRON-1752: Prevent package.lock from changing during build

## Contributor Comments
This PR updates the build script to use `npm ci` instead of `npm install`. The former command installs exactly what is in a project's package-lock.json file, whereas the later will reference package.json and update the lock file. Besides causing confusion with an unexpected file change, `npm install` had the potential to install different package versions of dependencies in different environments depending on when the install was last done. 

### Test Instructions
If you run a maven build with the current master branch, you will see package-lock.json update in the alerts ui. With this PR, there is no update to the lock file.

### Changes Included
* Update pom.xml files in both the alerts and management ui to use `npm ci` instead of `npm install`.
* Add a package-lock.json file to the management ui. There were no new dependencies added, just a locked version of dependencies that already existed in the project. This was done because the new command relies on reading the lock file to install exact versions of dependencies.


## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] N/A ~~Have you written or updated unit tests and or integration tests to verify your changes?~~
- [x] N/A ~~If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?~~

### For documentation related changes:
- [x] N/A ~~Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`~~:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
